### PR TITLE
refactor(configs): Update some default configs and remove some useless configs

### PR DESCRIPTION
**Pull request by acelyc111**
- Change some default config values in code base to make them more meaningful, including:

  | config name                                  | old default value | new default value |
  |----------------------------------------------|-------------------|-------------------|
  | log_private_reserve_max_size_mb              | 0                 | 1000              |
  | log_private_reserve_max_time_seconds         | 0                 | 36000             |
  | io_service_worker_count                      | 1                 | 4                 |
  | [threadpool..default]worker_count            | 2                 | 4                 |
  | [zookeeper]logfile                           | ""                | zoo.log           |
  | [tools.simple_logger]short_header            | true              | false             |
  | fd_lease_seconds                             | 9                 | 20                |
  | fd_grace_seconds                             | 10                | 22                |
  | replica_assign_delay_ms_for_dropouts         | 300000            | 600000            |
  | staleness_for_commit                         | 10                | 20                |
  | prepare_timeout_ms_for_secondaries           | 1000              | 3000              |
  | prepare_timeout_ms_for_potential_secondaries | 3000              | 5000              |
  | group_check_interval_ms                      | 10000             | 100000            |

- Change some config values in `src/server/config.ini` to keep consistent to code base, including:

  | config name                                          | old default value | new default value |
  |------------------------------------------------------|-------------------|-------------------|
  | [tools.simple_logger]max_number_of_log_files_on_disk | 500               | 20                |
  | [tools.simple_logger]stderr_start_level              | LOG_LEVEL_ERROR   | LOG_LEVEL_WARNING |

- Improve some config descriptions
- Remove some never used configs, including:
   - perf_counter_cluster_name
   - perf_counter_update_interval_seconds
   - perf_counter_enable_logging
   - perf_counter_enable_falcon
   - falcon_host
   - falcon_port
   - falcon_path
   - log_shared_file_size_mb
   - log_shared_gc_flush_replicas_limit
   - log_shared_batch_buffer_kb
   - log_shared_force_flush
   - log_shared_pending_size_throttling_threshold_kb
   - log_shared_pending_size_throttling_delay_ms
   - log_private_batch_buffer_kb
   - log_private_batch_buffer_count
   - log_private_batch_buffer_flush_interval_ms
   - verbose_log_on_commit
- Add a validator to check "fd_grace_seconds > fd_lease_seconds"
- Distinguish test and production environments

  | config name                   | test environment                | production environment             |
  |-------------------------------|---------------------------------|------------------------------------|
  | distributed_lock_service_type | distributed_lock_service_simple | distributed_lock_service_zookeeper |
  | meta_state_service_type       | meta_state_service_simple       | meta_state_service_zookeeper       |

### CODE REVIEW

File: src/replica/replica_stub.cpp [hunk 1]
```diff
@@ -242,8 +243,9 @@ DSN_DEFINE_bool(replication,
 DSN_DEFINE_bool(replication,
                 config_sync_disabled,
                 false,
-                "whether to disable replica configuration periodical sync with the meta server");
-DSN_DEFINE_bool(replication, fd_disabled, false, "whether to disable failure detection");
+                "The interval milliseconds of replica server to send replica config-sync "
+                "requests to meta server");
```
Comment (empiredan) [inline src/replica/replica_stub.cpp:269]:
```suggestion
                "Whether to disable replica server to send replica config-sync "
                "requests to meta server periodically");
```

File: src/replica/disk_cleaner.cpp [hunk 2]
```diff
@@ -38,11 +38,11 @@
...
 
-DSN_DEFINE_uint64(
-    replication,
-    gc_disk_error_replica_interval_seconds,
-    7 * 24 * 3600 /*7day*/,
-    "Duration of error replica being removed, which is in a directory with '.err' suffixed");
+DSN_DEFINE_uint64(replication,
+                  gc_disk_error_replica_interval_seconds,
+                  7 * 24 * 3600 /*7day*/,
+                  "The interval milliseconds to GC error replicas, which are in directories "
```
Comment (empiredan) [inline src/replica/disk_cleaner.cpp:44]:
```suggestion
                  "The interval in seconds to GC error replicas, which are in directories "
```

File: src/server/config.ini [hunk 3]
```diff
@@ -236,20 +236,20 @@ stateful = true
   delay_for_fd_timeout_on_start = false
   empty_write_disabled = false
 
-  prepare_timeout_ms_for_secondaries = 3000
-  prepare_timeout_ms_for_potential_secondaries = 5000
+  prepare_timeout_ms_for_secondaries = 1000
+  prepare_timeout_ms_for_potential_secondaries = 3000
```
Comment (empiredan) [inline src/server/config.ini:240]: Based on the experiences of production environment, sometimes both of them would be increased to 6000/10000. Thus I think 1000/3000 would be a litter small, which would make partitions unstable.
```suggestion
  prepare_timeout_ms_for_secondaries = 3000
  prepare_timeout_ms_for_potential_secondaries = 5000
```

File: src/server/config.ini [hunk 4]
```diff
@@ -206,7 +206,7 @@
   # partition guardian is used to keep partitions healthy. 
   # e.g. if max_replicas_in_group = 3, one primary and two secondaries must be kept for each partition 
   partition_guardian_type = partition_guardian
-  replica_assign_delay_ms_for_dropouts = 600000
+  replica_assign_delay_ms_for_dropouts = 300000
```
Comment (empiredan) [inline src/server/config.ini:209]: I think it's better to keep replica_assign_delay_ms_for_dropouts, staleness_for_commit, group_check_interval_ms, checkpoint_interval_seconds, fd_lease_seconds, fd_grace_seconds and log_private_reserve_max_time_seconds just as what they have been in `config.ini` ? All of them have been long running on production environments. I'm afraid that once they are changed, maybe there would be some problems for the production environment. We can change the default values in code to be consistent with what they are in `config.ini`.

File: src/server/config.ini (hunk 3 above)
Comment (acelyc111) [reply src/server/config.ini]: OK, I've reverted this change and updated the default values in code.

File: src/server/config.ini (hunk 4 above)
Comment (acelyc111) [reply src/server/config.ini]: OK, I've reverted these changes and updated the default values in code.

**Review by empiredan — DISMISSED** [review]

**Comment by empiredan** [comment]
Workflow `dsn.rep_tests.simple_kv` always failed for `case-603`:

```
./dsn.rep_tests.simple_kv case-603.ini case-603.act
Chosen stacktrace method is libgcc
Supported methods:
* generic_fp
* generic_fp_unsafe
* libgcc
* null

Use TCMALLOC_STACKTRACE_METHOD environment variable to override
W2024-02-08 05:08:05.613 (1707368885613654962 4511) : overwrite default thread pool for task RPC_CM_QUERY_PARTITION_CONFIG_BY_INDEX from THREAD_POOL_META_SERVER to THREAD_POOL_DEFAULT
W2024-02-08 05:08:05.613 (1707368885613684958 4511) : overwrite default thread pool for task RPC_CM_QUERY_PARTITION_CONFIG_BY_INDEX_ACK from THREAD_POOL_META_SERVER to THREAD_POOL_DEFAULT
   32  set:load_balance_for_test=1,not_exit_on_log_failure=1
    +  config:{0,-,[]}
    +  state:{{r1,ina,0,0,0}}
    +  config:{1,r1,[]}
    +  state:{{r1,pri,1,0,0}}
    +  state:{{r1,pri,1,0,0},{r2,ina,0,0,0}}
    +  state:{{r1,pri,1,0,0},{r2,pot,1,0,0}}
    +  state:{{r1,ina,1,0,0},{r2,pot,1,0,0}}
    +  config:{2,r1,[r2]}
    +  state:{{r1,pri,2,0,0},{r2,pot,1,0,0}}
    +  state:{{r1,pri,2,0,0},{r2,pot,1,0,0},{r3,ina,0,0,0}}
    +  state:{{r1,pri,2,0,0},{r2,pot,1,0,0},{r3,pot,2,0,0}}
    +  state:{{r1,ina,2,0,0},{r2,pot,1,0,0},{r3,pot,2,0,0}}
    +  config:{3,r1,[r2,r3]}
    +  state:{{r1,pri,3,0,0},{r2,pot,1,0,0},{r3,pot,2,0,0}}
    +  state:{{r1,pri,3,0,0},{r2,pot,1,0,0},{r3,sec,3,0,0}}
   35  state:{{r1,pri,3,0},{r2,sec,3,0},{r3,sec,3,0}}
   38  client:begin_write:id=1,key=k1,value=v1,timeout=0
   39  client:begin_write:id=2,key=k2,value=v2,timeout=0
   40  client:begin_write:id=3,key=k3,value=v3,timeout=0
   41  client:begin_write:id=4,key=k4,value=v4,timeout=0
   42  client:begin_write:id=5,key=k5,value=v5,timeout=0
   43  client:begin_write:id=6,key=k6,value=v6,timeout=0
    +  state:{{r1,pri,3,1,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
    +  state:{{r1,pri,3,2,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
    +  state:{{r1,pri,3,3,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
    +  state:{{r1,pri,3,4,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
    +  state:{{r1,pri,3,5,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
    s  client:end_write:id=6,err=ERR_OK,resp=0
    +  state:{{r1,pri,3,5,0},{r2,sec,3,1,0},{r3,sec,3,0,0}}
    s  client:end_write:id=2,err=ERR_OK,resp=0
    s  client:end_write:id=5,err=ERR_OK,resp=0
    s  client:end_write:id=4,err=ERR_OK,resp=0
    s  client:end_write:id=3,err=ERR_OK,resp=0
    +  state:{{r1,pri,3,5,0},{r2,sec,3,1,0},{r3,sec,3,1,0}}
    +  state:{{r1,pri,3,6,0},{r2,sec,3,1,0},{r3,sec,3,1,0}}
    s  client:end_write:id=1,err=ERR_OK,resp=0
    +  state:{{r1,pri,3,6,6},{r2,sec,3,1,0},{r3,sec,3,1,0}}
    +  state:{{r1,pri,3,6,6},{r2,sec,3,1,1},{r3,sec,3,1,0}}
   46  state:{{r1,pri,3,6,6},{r2,sec,3,1,1},{r3,sec,3,1,1}}
   49  set:disable_load_balance=1
   51  client:replica_config:receiver=r1,type=downgrade_to_inactive,node=r1
    +  state:{{r1,ina,3,6,6},{r2,sec,3,1,1},{r3,sec,3,1,1}}
   52  config:{4,-,[r2,r3]}
    +  state:{{r1,ina,4,6,6},{r2,sec,3,1,1},{r3,sec,3,1,1}}
    +  state:{{r1,ina,4,6,6},{r2,sec,4,1,1},{r3,sec,3,1,1}}
    +  state:{{r1,err,4,6,6},{r2,sec,4,1,1},{r3,sec,3,1,1}}
    +  state:{{r2,sec,4,1,1},{r3,sec,3,1,1}}
    +  state:{{r2,sec,4,1,1},{r3,sec,4,1,1}}
   53  state:{{r1,ina,4,6},{r2,sec,4,1},{r3,sec,4,1}}
 <==>  null_loop:10000
dsn exit with code -1
run case-603 failed, return value = 255
Error: Process completed with exit code 1.
```

**Comment by acelyc111** [comment]
> Workflow `dsn.rep_tests.simple_kv` always failed for `case-603`:
> 
> ```
> ./dsn.rep_tests.simple_kv case-603.ini case-603.act
> Chosen stacktrace method is libgcc
> Supported methods:
> * generic_fp
> * generic_fp_unsafe
> * libgcc
> * null
> 
> Use TCMALLOC_STACKTRACE_METHOD environment variable to override
> W2024-02-08 05:08:05.613 (1707368885613654962 4511) : overwrite default thread pool for task RPC_CM_QUERY_PARTITION_CONFIG_BY_INDEX from THREAD_POOL_META_SERVER to THREAD_POOL_DEFAULT
> W2024-02-08 05:08:05.613 (1707368885613684958 4511) : overwrite default thread pool for task RPC_CM_QUERY_PARTITION_CONFIG_BY_INDEX_ACK from THREAD_POOL_META_SERVER to THREAD_POOL_DEFAULT
>    32  set:load_balance_for_test=1,not_exit_on_log_failure=1
>     +  config:{0,-,[]}
>     +  state:{{r1,ina,0,0,0}}
>     +  config:{1,r1,[]}
>     +  state:{{r1,pri,1,0,0}}
>     +  state:{{r1,pri,1,0,0},{r2,ina,0,0,0}}
>     +  state:{{r1,pri,1,0,0},{r2,pot,1,0,0}}
>     +  state:{{r1,ina,1,0,0},{r2,pot,1,0,0}}
>     +  config:{2,r1,[r2]}
>     +  state:{{r1,pri,2,0,0},{r2,pot,1,0,0}}
>     +  state:{{r1,pri,2,0,0},{r2,pot,1,0,0},{r3,ina,0,0,0}}
>     +  state:{{r1,pri,2,0,0},{r2,pot,1,0,0},{r3,pot,2,0,0}}
>     +  state:{{r1,ina,2,0,0},{r2,pot,1,0,0},{r3,pot,2,0,0}}
>     +  config:{3,r1,[r2,r3]}
>     +  state:{{r1,pri,3,0,0},{r2,pot,1,0,0},{r3,pot,2,0,0}}
>     +  state:{{r1,pri,3,0,0},{r2,pot,1,0,0},{r3,sec,3,0,0}}
>    35  state:{{r1,pri,3,0},{r2,sec,3,0},{r3,sec,3,0}}
>    38  client:begin_write:id=1,key=k1,value=v1,timeout=0
>    39  client:begin_write:id=2,key=k2,value=v2,timeout=0
>    40  client:begin_write:id=3,key=k3,value=v3,timeout=0
>    41  client:begin_write:id=4,key=k4,value=v4,timeout=0
>    42  client:begin_write:id=5,key=k5,value=v5,timeout=0
>    43  client:begin_write:id=6,key=k6,value=v6,timeout=0
>     +  state:{{r1,pri,3,1,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
>     +  state:{{r1,pri,3,2,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
>     +  state:{{r1,pri,3,3,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
>     +  state:{{r1,pri,3,4,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
>     +  state:{{r1,pri,3,5,0},{r2,sec,3,0,0},{r3,sec,3,0,0}}
>     s  client:end_write:id=6,err=ERR_OK,resp=0
>     +  state:{{r1,pri,3,5,0},{r2,sec,3,1,0},{r3,sec,3,0,0}}
>     s  client:end_write:id=2,err=ERR_OK,resp=0
>     s  client:end_write:id=5,err=ERR_OK,resp=0
>     s  client:end_write:id=4,err=ERR_OK,resp=0
>     s  client:end_write:id=3,err=ERR_OK,resp=0
>     +  state:{{r1,pri,3,5,0},{r2,sec,3,1,0},{r3,sec,3,1,0}}
>     +  state:{{r1,pri,3,6,0},{r2,sec,3,1,0},{r3,sec,3,1,0}}
>     s  client:end_write:id=1,err=ERR_OK,resp=0
>     +  state:{{r1,pri,3,6,6},{r2,sec,3,1,0},{r3,sec,3,1,0}}
>     +  state:{{r1,pri,3,6,6},{r2,sec,3,1,1},{r3,sec,3,1,0}}
>    46  state:{{r1,pri,3,6,6},{r2,sec,3,1,1},{r3,sec,3,1,1}}
>    49  set:disable_load_balance=1
>    51  client:replica_config:receiver=r1,type=downgrade_to_inactive,node=r1
>     +  state:{{r1,ina,3,6,6},{r2,sec,3,1,1},{r3,sec,3,1,1}}
>    52  config:{4,-,[r2,r3]}
>     +  state:{{r1,ina,4,6,6},{r2,sec,3,1,1},{r3,sec,3,1,1}}
>     +  state:{{r1,ina,4,6,6},{r2,sec,4,1,1},{r3,sec,3,1,1}}
>     +  state:{{r1,err,4,6,6},{r2,sec,4,1,1},{r3,sec,3,1,1}}
>     +  state:{{r2,sec,4,1,1},{r3,sec,3,1,1}}
>     +  state:{{r2,sec,4,1,1},{r3,sec,4,1,1}}
>    53  state:{{r1,ina,4,6},{r2,sec,4,1},{r3,sec,4,1}}
>  <==>  null_loop:10000
> dsn exit with code -1
> run case-603 failed, return value = 255
> Error: Process completed with exit code 1.
> ```

Fixed

**Review by empiredan — APPROVED** [review]

**Review by GehaFearless — APPROVED** [review]

**Pull request merged**